### ISSUE TITLE
CA1052 false positive

### ISSUE THREAD

**Issue by kentcb** [dotnet/roslyn-analyzers#1320]
#### Analyzer package

Microsoft.CodeAnalysis.FxCopAnalyzers

#### Analyzer

CA1052

#### Repro steps

```C#
public sealed class EmptyDto
{
    public static readonly EmptyDto Instance = new EmptyDto();

    private EmptyDto()
    {
    }
}
```

#### Expected behavior

No violation (because the type is sealed).

#### Actual behavior

```
warning CA1052: Type 'EmptyDto' is a static holder type but is neither static nor NotInheritable
```

**Comment by mavasani** [comment]
Verified this is not reported by legacy FxCop (see internal implementation [here](http://index/?query=StaticHolderTypesShouldBeSealed&rightProject=FxCopSdk&file=InternalUtilities.cs&line=919) and is likely a common coding style. This should be fixed in 15.5

**Comment by 333fred** [comment]
Note that the reason that a diagnostic is not reported here is _not_ because the type is sealed. The original FxCop implementation did not report an error because of the private constructor. A note from the original implementation:
```C#
// Before the 'static' class modifier was added, it
// was common to add a private constructor to a static
// holder type so that it couldn't be created.
```
